python/GafferSceneUI/LightEditor.py

```diff
@@ -352,6 +353,69 @@ def __editSelectedCells( self, pathListing ) :
 
 			self.__popup.popup()
 
+	def __buttonPress( self, pathListing, event ) :
+
+		if event.button != event.Buttons.Right or event.modifiers != event.Modifiers.None_ :
+			return False
+
+		selection = pathListing.getSelection()
+
+		columns = pathListing.getColumns()
+		cellColumn = pathListing.columnAt( event.line.p0 )
+		columnIndex = -1
+		for i in range( 0, len( columns ) ) :
+			if cellColumn == columns[i] :
+				columnIndex = i
+
+		if columnIndex == -1 :
+			return False
+
+		cellPath = pathListing.pathAt( event.line.p0 )
+
+		if str( cellPath ) not in selection[columnIndex].paths() :
+			for p in selection :
+				p.clear()
+			selection[columnIndex].addPath( str( cellPath ) )
+			pathListing.setSelection( selection )
+
+		menuDefinition = IECore.MenuDefinition()
+
+		menuDefinition.append(
+			"Show History...",
+			{
+				"command" : Gaffer.WeakMethod( self.__showHistory ),
+				"active" : any( i.paths() for i in selection )
```

Comment:
This is turning the selection into a list - another linear time operation. We can use `not i.isEmpty()` for a constant-time result.

Reply:
Fixed in 6d30a4c7b7d4671509b29ff57937016d63252b30